src/bp-messages/classes/class-bp-messages-thread.php

```diff
@@ -945,6 +956,43 @@ public static function get_total_threads_for_user( $user_id, $box = 'inbox', $ty
 		return (int) $wpdb->get_var( $wpdb->prepare( "SELECT COUNT(thread_id) FROM {$bp->messages->table_name_recipients} WHERE user_id = %d AND is_deleted = 0 {$exclude_sender} {$type_sql}", $user_id ) );
 	}
 
+	/**
+	 * Returns the total number of messages in a thread.
+	 *
+	 * @since 12.0.0
+	 *
+	 * @global BuddyPress $bp The one true BuddyPress instance.
+	 * @global wpdb $wpdb WordPress database object.
+	 *
+	 * @param int $thread_id The message thread ID.
+	 * @return int Total thread message count
+	 */
+	public static function get_total_thread_message_count( $thread_id ) {
+		global $wpdb;
+
+		$total_count = wp_cache_get( $thread_id, 'bp_messages_unread_count' );
```

Comment:
I think we should use another cache group, as this one is used to get the unread count. This function, if I understand well, gets the amount of messages for a thread no matter if messages are unread or read. About the cache group name, something like `bp_messages_thread_count` might be suitable...

NB: You'll need to also add the new cache group to the `BP_Messages_Component::setup_cache_groups()`

Reply:
Ops! Good cache! This is a mistake!